cmd/soroban-cli/src/fee.rs

```diff
@@ -16,15 +17,31 @@ pub struct Args {
     /// Number of instructions to simulate
     #[arg(long, help_heading = HEADING_RPC)]
     pub instructions: Option<u32>,
+    /// Build the transaction only write the base64 xdr to stdout
+    #[arg(long, help_heading = HEADING_RPC)]
+    pub build_only: bool,
+    /// Simulation the transaction only write the base64 xdr to stdout
+    #[arg(long, help_heading = HEADING_RPC, conflicts_with = "build_only")]
+    pub sim_only: bool,
 }
 
 impl Args {
-    pub fn apply_to_assembled_txn(&self, txn: Assembled) -> Assembled {
-        if let Some(instructions) = self.instructions {
+    pub fn apply_to_assembled_txn(&self, txn: Assembled) -> Result<Assembled, xdr::Error> {
+        let simulated_txn = if let Some(instructions) = self.instructions {
             txn.set_max_instructions(instructions)
         } else {
             add_padding_to_instructions(txn)
+        };
+        if self.sim_only {
+            println!(
+                "{}",
+                simulated_txn
+                    .transaction()
+                    .to_xdr_base64(xdr::Limits::none())?
+            );
+            std::process::exit(0);
```

Comment:
It's difficult to follow the application flow because of this exit buried in the code. And the code path of the --build-only option doesn't result in this. Is this something that's too difficult to change?

Reply:
Nope! This should use the same `TxnResult` type since it has the notion of item vs base64 xdr. Will update!